src/server/multiplayer/MultiplayerServerState.js

```diff
@@ -1,25 +1,36 @@
 /* eslint-disable no-console */
-import { compact, find, pull, reject } from 'lodash';
+import { chunk, compact, find, mapValues, pull, reject } from 'lodash';
 
 import { id as generateID } from '../../common/util/common';
+import { saveGame } from '../../common/util/firebase';
+import defaultGameState from '../../common/store/defaultGameState';
+import gameReducer from '../../common/reducers/game';
 
 import { getPeopleInGame, withoutClient } from './util';
 
 export default class MultiplayerServerState {
   state = {
     connections: {},  // map of { clientID: websocket }
     games: [],  // array of { id, name, format, players, playerColors, spectators, actions, decks, usernames, startingSeed }
+    gameObjects: {}, // map of { gameID: game }
     waitingPlayers: [],  // array of { id, name, format, deck, players }
+    matchmakingQueue: [], // array of { clientID, deck }
     playersOnline: [],  // array of clientIDs
-    usernames: {}  // map of { clientID: username }
+    userData: {} // map of { clientID: { uid, displayName, ... } }
```

Comment:
I wonder if the `{ uid, displayName, ... }` object should be a `User` class or something

Reply:
Let's do this later, preferably once we get Typescript on the server.